core/src/main/scala/slamdata/AutoBump.scala

```diff
@@ -0,0 +1,369 @@
+      _ <- close(pullRequest, s"${pullRequest.title} (preceded by #${oldest.number})")
+      _ <- IO(log.info(s"Closed $owner/$repoSlug#${pullRequest.number}, preceded by $owner/$repoSlug#${oldest.number}"))
+      _ <- deleteBranch(pullRequest)
+      _ <- IO(log.info(s"Removed branch ${pullRequest.base.map(_.ref)} from $owner/$repoSlug"))
+    } yield NotOldest(oldest, pullRequest).asLeft[PullRequestDraft]
+
+    // TODO: Update pull request description?
+    def createOrUpdatePullRequest(branchName: String,
+                                  changes: List[String],
+                                  changeLabel: ChangeLabel,
+                                  maybePullRequest: Option[PullRequestDraft]): IO[Either[Warnings, PullRequestDraft]] = {
+      for {
+        pullRequest <- maybePullRequest.fold {
+          draftPullRequest(authorRepository, branchName, changes.mkString("\n"))
+            .flatTap(pullRequest => IO(log.info(s"Opened $owner/$repoSlug#${pullRequest.number}")))
+        }(IO.pure)
+        labels <- getLabels(pullRequest.number)
+        prChangeLabels = labels.flatMap(label => ChangeLabel(label.name)).toSet
+        highestChange = (prChangeLabels + changeLabel).max
+        lowerChanges = (prChangeLabels - highestChange).toList
+        _ <- if (prChangeLabels.contains(highestChange)) IO.unit else assignLabel(highestChange.label, pullRequest)
+        _ <- lowerChanges.traverse(change => removeLabel(pullRequest, change.label))
+        oldestPullRequest <- getOldestAutoBumpPullRequest
+        res <- oldestPullRequest
+          .filter(_.number == pullRequest.number)
+          .fold(ifOldest(pullRequest))(ifNotOldest(_, pullRequest))
+      } yield res
+    }
+
+    val app = for {
+      (dir, branchName, maybePullRequest, changes, label) <- EitherT(tryUpdateDependencies)
+      _ <- EitherT(tryCommit(dir))
+      _ <- EitherT(tryPush(dir, branchName))
+      pullRequest <- EitherT(createOrUpdatePullRequest(branchName, changes, label, maybePullRequest))
+    } yield pullRequest
+
+    app.leftSemiflatMap(_.warn(log)).value.unsafeRunSync().isRight
+  }
+
+  private def run(command: String,
+                  log: Logger,
+                  merge: Boolean = false,
+                  workingDir: Option[File] = None): IO[List[String]] = {
+    run(command.split("""\s+""").toVector, log, merge, workingDir)
+  }
+
+  private def run(command: Seq[String],
+                  log: Logger,
+                  merge: Boolean,
+                  workingDir: Option[File],
+                  env: (String, String)*): IO[List[String]] = {
+    val lines = mutable.Buffer[String]()
+    val processLogger = getProcessLogger(log, merge, lines)
+    run_?(command, log, processLogger, workingDir, env: _*)
+      .ensureOr(res => new RuntimeException(s"${command.take(2).mkString(" ")} exit code $res"))(_ == 0)
+      .as(lines.toList)
+  }
+
+  private def run_?(command: Seq[String],
```

Comment:
Suffix operators on alphabetical identifiers. Now there's a style I haven't seen [in a long, long time…](https://github.com/lift/framework)

Reply:
Nothing else occurred to me, I'm afraid. It felt wrong and icky, but also understandable. I'm still a bit disgruntled that `!` and `?` cannot appear as suffixes to alphabetical identifiers, ten years later. :)